src/ipc/ipc4/handler.c

```diff
@@ -630,6 +630,15 @@ static int ipc4_set_pipeline_state(struct ipc4_message_request *ipc4)
 			ipc_compound_pre_start(state.primary.r.type);
 			ret = ipc4_pipeline_trigger(ppl_icd, cmd, &delayed);
 			ipc_compound_post_start(state.primary.r.type, ret, delayed);
+			if (delayed) {
+				/* To maintain pipeline order for triggers,
+				 * must wait until trigger action is done
+				 */
+				if (ipc_wait_for_compound_msg() != 0) {
+					ipc_cmd_err(&ipc_tr, "ipc4: fail with delayed trigger");
+					return IPC4_FAILURE;
+				}
+			}
```

Comment:
hmm, so that means that in your example the IPC thread might be delayed by up to 3 (or more for more pipelines) ms? I guess it won't break anything, but maybe at least mention this in the comment above?

Reply:
@lyakh Ack. I extended the comment to explain this, see updated patch.